R/Data-methods.R

```diff
@@ -560,3 +560,50 @@ setMethod(
     eff
   }
 )
+
+# ngrid ----
+
+## generic ----
+
+#' Number of Doses in Grid
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A function that gets the number of doses in grid. User can choose whether
+#' the placebo dose (if any) should be counted or not.
+#'
+#' @param object (`Data`)\cr object with dose grid.
+#' @param ... further arguments passed to class-specific methods.
+#' @return `integer` the number of doses in grid.
+#' @export
+#'
+setGeneric(
+  name = "ngrid",
+  def = function(object, ...) {
```

Comment:
maybe consider moving the ignore argument to the generic, in case all methods should have it

Reply:
Tx! Should the ignore param be used in signature or just a non-signature param?

Reply:
agree